airflow-core/src/airflow/ui/src/pages/TaskInstances/TaskInstancesFilter.tsx

```diff
@@ -101,66 +191,106 @@ export const TaskInstancesFilter = ({
+        />
+      </HStack>
+      <HStack>
+        <AttrSelectFilterMulti
+          displayPrefix={undefined}
+          handleSelect={handleSelectedOperators}
+          placeholderText={translate("selectOperator")}
+          selectedValues={selectedOperators}
+          values={allOperatorNames}
+        />
+        <AttrSelectFilterMulti
+          displayPrefix={undefined}
+          handleSelect={handleSelectedQueues}
+          placeholderText={translate("selectQueues")}
+          selectedValues={selectedQueues}
+          values={allQueueValues}
+        />
+        <AttrSelectFilterMulti
+          displayPrefix={undefined}
+          handleSelect={handleSelectedPools}
+          placeholderText={translate("selectPools")}
+          selectedValues={selectedPools}
+          values={allPoolValues}
+        />
+        <Box>
+          <ResetButton filterCount={taskFilterCount} onClearFilters={onClearFilters} />
+        </Box>
+      </HStack>
+      <VStack alignItems="flex-start" gap={1}>
+        <FilterBar
+          configs={filterConfigs}
+          initialValues={initialValues}
+          onFiltersChange={handleFiltersChange}
         />
-      )}
-      <SearchBar
-        buttonProps={{ disabled: true }}
-        defaultValue={taskDisplayNamePattern ?? ""}
-        hideAdvanced
-        hotkeyDisabled={Boolean(runId)}
-        onChange={handleSearchChange}
-        placeHolder={translate("dags:search.tasks")}
-      />
-      <Select.Root
-        collection={taskInstanceStateOptions}
-        maxW="450px"
-        multiple
-        onValueChange={handleStateChange}
-        value={hasFilteredState ? filteredState : ["all"]}
-      >
-        <Select.Trigger
-          {...(hasFilteredState ? { clearable: true } : {})}
-          colorPalette="brand"
-          isActive={Boolean(filteredState)}
-        >
-          <Select.ValueText>
-            {() =>
-              hasFilteredState ? (
-                <HStack flexWrap="wrap" fontSize="sm" gap="4px" paddingY="8px">
-                  {filteredState.map((state) => (
-                    <StateBadge key={state} state={state as TaskInstanceState}>
-                      {translate(`common:states.${state}`)}
-                    </StateBadge>
-                  ))}
-                </HStack>
-              ) : (
-                translate("dags:filters.allStates")
-              )
-            }
-          </Select.ValueText>
-        </Select.Trigger>
-        <Select.Content>
-          {taskInstanceStateOptions.items.map((option) => (
-            <Select.Item item={option} key={option.label}>
-              {option.value === "all" ? (
-                translate(option.label)
-              ) : (
-                <StateBadge state={option.value as TaskInstanceState}>{translate(option.label)}</StateBadge>
-              )}
-            </Select.Item>
-          ))}
-        </Select.Content>
-      </Select.Root>
-    </HStack>
+      </VStack>
```

Comment:
We want to move all of the filters into FilterBar. Then they don't take up so much room when they're not being used.